staging/src/k8s.io/apiserver/pkg/storageversion/updater.go

```diff
@@ -121,13 +124,19 @@ func setStatusCondition(conditions *[]v1alpha1.StorageVersionCondition, newCondi
 	existingCondition.ObservedGeneration = newCondition.ObservedGeneration
 }
 
-// updateStorageVersionFor updates the storage version object for the resource.
-func updateStorageVersionFor(c Client, apiserverID string, gr schema.GroupResource, encodingVersion string, decodableVersions []string) error {
+// UpdateStorageVersionFor updates the storage version object for the resource.
+func UpdateStorageVersionFor(ctx context.Context, c Client, apiserverID string, gr schema.GroupResource, encodingVersion string, decodableVersions []string, postProcessFunc processStorageVersionFunc) error {
```

Comment:
doc what postProcessFunc is and when it is called. I assume it is called before updating the status of the CRD.

Reply:
it modifies the StorageVersion object before the object gets created/updated. It allows customizing the object (e.g. setting an owner reference pointing to a CRD) as a last step